include/lin/references/vector_mapping_reference.hpp

```diff
@@ -0,0 +1,202 @@
+// vim: set tabstop=2:softtabstop=2:shiftwidth=2:expandtab
+
+/** @file lin/references/vector_mapping_reference.hpp
+ *  @author Kyle Krol
+ */
+
+#ifndef LIN_REFERENCES_VECTOR_MAPPING_REFERENCE_HPP_
+#define LIN_REFERENCES_VECTOR_MAPPING_REFERENCE_HPP_
+
+#include "../core.hpp"
+#include "tensor_mapping_reference.hpp"
+
+namespace lin {
+namespace internal {
+
+/** @brief Generic vector reference with read and write access.
+ *
+ *  @tparam E  Underlying referenced type.
+ *  @tparam N  Length at compile time.
+ *  @tparam MN Maximum length at compile time.
+ *
+ *  The template parameters specify the reference's traits. The traits must
+ *  qualify the type as a column vector.
+ *
+ *  It's important to note, if the underlying type goes out of scope the
+ *  reference is invalidated.
+ *
+ *  @sa internal::TensorMappingReference
+ *  @sa internal::traits
+ *  @sa internal::is_col_vector
+ *
+ *  @ingroup REFERENCES
+ */
+template <class E, size_t N, size_t MN>
+class VectorMappingReference : public TensorMappingReference<VectorMappingReference<E, N, MN>, E> {
+  static_assert(is_col_vector<VectorMappingReference<E, N, MN>>::value,
+      "Derived types to VectorMappingReference<...> must have column vector traits");
+
+ public:
+  /** @brief Traits information for this type.
+   *
+   *  @sa internal::traits
+   */
+  typedef traits<VectorMappingReference<E, N, MN>> Traits;
+
+  /** @brief %Vector traits information for this type.
+   *
+   *  @sa internal::vector_traits
+   */
+  typedef vector_traits<VectorMappingReference<E, N, MN>> VectorTraits;
+
+ protected:
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::derived;
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::resize;
+
+ public:
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::TensorMappingReference;
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::rows;
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::cols;
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::size;
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::operator=;
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::operator();
+  using TensorMappingReference<VectorMappingReference<E, N, MN>, E>::eval;
+
+  constexpr VectorMappingReference() = delete;
+  constexpr VectorMappingReference(VectorMappingReference<E, N, MN> const &) = default;
+  constexpr VectorMappingReference(VectorMappingReference<E, N, MN> &&) = default;
+  constexpr VectorMappingReference<E, N, MN> &operator=(VectorMappingReference<E, N, MN> const &) = default;
+  constexpr VectorMappingReference<E, N, MN> &operator=(VectorMappingReference<E, N, MN> &&) = default;
+
+  /** @brief Constructs a new vector reference with the provided mapping, anchor
```

Comment:
is this col vector by default? I see a row vector below, but no col vector?

Reply:
Sorry perhaps i haven't clarified by a column vector is always the standard vector type and row vector always has the added modifier row. Notice the static assert on line 36 ensures this type always represents a column vector.

Reply:
nice bet